build-scripts/generate-bom.py

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env python3
+
+import json
+import os
+from pathlib import Path
+import subprocess
+
+DIR = Path(__file__).absolute().parent
+
+SNAPCRAFT_PART_BUILD = Path(os.getenv("SNAPCRAFT_PART_BUILD", ""))
+SNAPCRAFT_PART_INSTALL = Path(os.getenv("SNAPCRAFT_PART_INSTALL", ""))
+
+BUILD_DIRECTORY = SNAPCRAFT_PART_BUILD.exists() and SNAPCRAFT_PART_BUILD or DIR / ".build"
+INSTALL_DIRECTORY = SNAPCRAFT_PART_INSTALL.exists() and SNAPCRAFT_PART_INSTALL or DIR / ".install"
+
+TOOLS = {
+    "go": ["go", "version"],
+    "gcc": ["gcc", "--version"],
+}
+
+COMPONENTS = [
```

Comment:
This is a good idea but seemed to make things worse in the end. Some of the components are not in use (e.g. crictl is not built into the snap at all), and others are not useful because they are internal only and do not make sense (e.g. the microk8s-completion component uses helm to only generate the autocompletion script).

Up to you to update this

Reply:
Initially I retrieved this from the parts/bom/after in snapcraft.yaml, but requires python3-yaml build package which i would rather avoid adding as a build package to not mess with the environment

Reply:
I would second trying to get the list from the directory and having like a `.bomignore` to exclude any internal files we want. It sounds easier to maintain and not forget something in the bom.